packages/api/amplify_api/example/integration_test/graphql/api_key_test.dart

```diff
@@ -23,73 +25,80 @@ import '../util.dart';
-          if (!useExistingTestUser) {
-            await deleteTestUser();
-          }
-        });
-
-        testWidgets(
-            'should emit event when onCreate subscription made with model helper',
+      group('queries', () {
+        testWidgets('should query authorized model',
             (WidgetTester tester) async {
-          String name = 'Integration Test Blog - subscription create ${uuid()}';
-          final subscriptionRequest = ModelSubscriptions.onCreate(
+          final req = ModelQueries.list<Blog>(
             Blog.classType,
             authorizationMode: APIAuthorizationType.apiKey,
           );
+          final res = await Amplify.API.query(request: req).response;
+          final data = res.data;
+          expect(res, hasNoGraphQLErrors);
+          expect(data?.items.length, greaterThanOrEqualTo(0));
+        });
 
-          final eventResponse = await establishSubscriptionAndMutate(
-            subscriptionRequest,
-            () => addBlog(name),
-            eventFilter: (Blog? blog) => blog?.name == name,
+        testWidgets('should get error for unauthorized model', (
+          WidgetTester tester,
+        ) async {
+          final req = ModelQueries.list<Post>(
+            Post.classType,
+            authorizationMode: APIAuthorizationType.apiKey,
           );
-          Blog? blogFromEvent = eventResponse.data;
-
-          expect(blogFromEvent?.name, equals(name));
+          final res = await Amplify.API.query(request: req).response;
+          expect(res.hasErrors, true);
+          expect(res.data, isNull);
         });
-      },
-    );
-  });
+      });
+
+      group(
+        'subscriptions',
+        () {
+          setUpAll(() async {
+            if (!useExistingTestUser) {
+              await signUpTestUser();
+            }
+            await signInTestUser();
+          });
+
+          tearDownAll(() async {
+            await deleteTestModels();
+            if (!useExistingTestUser) {
+              await deleteTestUser();
+            }
+          });
+
+          testWidgets(
+              'should emit event when onCreate subscription made with model helper',
+              (WidgetTester tester) async {
+            String name =
+                'Integration Test Blog - subscription create ${uuid()}';
+            final subscriptionRequest = ModelSubscriptions.onCreate(
+              Blog.classType,
+              authorizationMode: APIAuthorizationType.apiKey,
+            );
+
+            final eventResponse = await establishSubscriptionAndMutate(
+              subscriptionRequest,
+              () => addBlog(name),
+              eventFilter: (Blog? blog) => blog?.name == name,
+            );
+            Blog? blogFromEvent = eventResponse.data;
+
+            expect(blogFromEvent?.name, equals(name));
+          });
+        },
+        // TODO(equartey): ensure state machine GQL sub impl doesn't have this issue.
+        skip: Platform.isWindows,
```

Comment:
Will this work on web? You may need
```suggestion
        skip: !zIsWeb && Platform.isWindows,
```

Reply:
Instead of skipping these groups you can add a `TestOn` annotation to the top of the files (above imports):

```dart
@TestOn('!windows')
```

For future reference- not sure you need to change here.

Reply:
thanks for tips, added `!zIsWeb` to these